lib/widgets/inventory/inventory_browser_app_bar.dart

```diff
@@ -19,32 +21,16 @@ class InventoryBrowserAppBar extends StatefulWidget {
 }
 
 class _InventoryBrowserAppBarState extends State<InventoryBrowserAppBar> {
-  final ReceivePort _port = ReceivePort();
-
-  @override
-  void initState() {
-    super.initState();
-
-    IsolateNameServer.registerPortWithName(_port.sendPort, 'downloader_send_port');
-    _port.listen((dynamic data) {
-      // Not useful yet? idk...
-      // String id = data[0];
-      // DownloadTaskStatus status = data[1];
-      // int progress = data[2];
-    });
-
-    FlutterDownloader.registerCallback(downloadCallback);
-  }
-
-  @override
-  void dispose() {
-    IsolateNameServer.removePortNameMapping('downloader_send_port');
-    super.dispose();
-  }
+  final Future<Directory> _tempDirectoryFuture = getTemporaryDirectory();
 
   @pragma('vm:entry-point')
-  static void downloadCallback(String id, int status, int progress) {
-    final SendPort? send = IsolateNameServer.lookupPortByName('downloader_send_port');
+  static void downloadCallback(TaskUpdate event) {
```

Comment:
Do we need both this entry point and the one in main.dart? Seems one of these might be redundant now

Reply:
They very much are, but it won't compile without both, as the declaration for the listener needs to see it. However, when it's called in the context of the inventory browser, it needs to actually exist there.

Reply:
Gotcha, makes sense